### PR TITLE
Fix arima

### PR DESCRIPTION
**Pull request by guilhermebodin**

### CODE REVIEW

File: src/hyperparameters.jl [hunk 1]
```diff
@@ -198,7 +198,10 @@ function set_initial_hyperparameters!(
     model::StateSpaceModel, initial_hyperparameters::Dict{String,<:Real}
 )
     for (k, v) in initial_hyperparameters
-        update_constrained_value!(model, k, v)
+        # Only update if it has not been updated before.
+        if isnan(get_constrained_value(model, k))
+            update_constrained_value!(model, k, v)
+        end
```
Comment (raphaelsaavedra) [inline src/hyperparameters.jl:204]:
```suggestion
        isnan(get_constrained_value(model, k)) && update_constrained_value!(model, k, v)
```
Can do it in one line

File: src/models/arima.jl [hunk 2]
```diff
@@ -162,8 +162,9 @@ function impose_unit_root_constraint(hyperparameter_values::Vector{Fl}) where Fl
     n = length(hyperparameter_values)
     y = fill(zero(Fl), n, n)
     r = @. hyperparameter_values / sqrt((1 + hyperparameter_values^2))
+    # TODO must check but I am almost sure (Guilherme)
```
Comment (raphaelsaavedra) [inline src/models/arima.jl:165]: Want to wait until you're sure before we merge?

File: src/hyperparameters.jl (hunk 1 above)
Comment (guilhermebodin) [reply src/hyperparameters.jl:204]: I think the one above is more clear...

File: src/models/arima.jl (hunk 2 above)
Comment (guilhermebodin) [reply src/models/arima.jl:165]: Now I am sure

**Comment by guilhermebodin** [comment]
My heuristic failed on 1.5 and I don`t know exactly why

**Comment by raphaelsaavedra** [comment]
There is a simulation in the test, so there is some randomness. The RNG changed in 1.5, so it might be that. Maybe just try changing the seed?

**Comment by guilhermebodin** [comment]
The problem was the heuristic itself, I took one from other ARIMA packages now.

**Review by raphaelsaavedra — CHANGES_REQUESTED** [review]
Add compat entry for `ShiftedArrays` and I think it's all good

**Pull request merged**